// react-leaflet-google/lib/GoogleLayer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _leaflet = require("leaflet");

var L = _interopRequireWildcard(_leaflet);

var _reactLeaflet = require("react-leaflet");

require("./leaflet.google");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var GoogleLayer = function (_GridLayer) {
  _inherits(GoogleLayer, _GridLayer);

  function GoogleLayer() {
    _classCallCheck(this, GoogleLayer);

    return _possibleConstructorReturn(this, (GoogleLayer.__proto__ || Object.getPrototypeOf(GoogleLayer)).apply(this, arguments));
  }

  _createClass(GoogleLayer, [{
    key: "createLeafletElement",
    value: function createLeafletElement(props) {
      this.leafletElement = new L.gridLayer.googleMutant(props);
      return this.leafletElement;
    }
  }, {
    key: "updateLeafletElement",
    value: function updateLeafletElement(prevProps, nextProps) {
      var opacity = nextProps.opacity,
          zIndex = nextProps.zIndex;

      if (opacity !== prevProps.opacity) {
        this.leafletElement.setOpacity(opacity);
      }
      if (zIndex !== prevProps.zIndex) {
        this.leafletElement.setZIndex(zIndex);
      }
    }
  }]);

  return GoogleLayer;
}(_reactLeaflet.GridLayer);

GoogleLayer.propTypes = {
  googlekey: _propTypes2.default.string.isRequired,
  maptype: _propTypes2.default.string,
  asclientid: _propTypes2.default.bool,
  channel: _propTypes2.default.string
};
GoogleLayer.contextTypes = _reactLeaflet.GridLayer.contextTypes;
GoogleLayer.childContextTypes = _reactLeaflet.GridLayer.childContextTypes;
exports.default = (0, _reactLeaflet.withLeaflet)(GoogleLayer);
